### PR TITLE
CDAP-4470 HttpRequests to reuse connections.

### PR DESCRIPTION
**Pull request by anwar6953**
Update HttpRequests.java to reuse connections across calls to its execute method.
The updated test case in this PR checks that the connection is reused.

https://issues.cask.co/browse/CDAP-4470

### CODE REVIEW

**Comment by anwar6953** [comment]
Build was failing, so I removed dependency on repository.continuuity.com.
Related: https://github.com/caskdata/cdap/pull/4559

File: common-http/src/main/java/co/cask/common/http/HttpRequests.java [hunk 1]
```diff
@@ -114,8 +115,11 @@ public static HttpResponse execute(HttpRequest request, HttpRequestConfig reques
 
       try {
         if (isSuccessful(conn.getResponseCode())) {
-          return new HttpResponse(conn.getResponseCode(), conn.getResponseMessage(),
-                                  ByteStreams.toByteArray(conn.getInputStream()), conn.getHeaderFields());
+          HttpResponse response =
+            new HttpResponse(conn.getResponseCode(), conn.getResponseMessage(),
+                             ByteStreams.toByteArray(conn.getInputStream()), conn.getHeaderFields());
+          conn.getInputStream().close();
```
Comment (chtyim) [inline common-http/src/main/java/co/cask/common/http/HttpRequests.java]: Don't call `getInputStream()` multiple times. Use the try-resource construct:

``` java
try (InputStream is = conn.getInputStream()) {
  return new HttpResponse(...., ByteStreams.toByteArray(is), ....);
}
```

Comment (anwar6953) [inline common-http/src/main/java/co/cask/common/http/HttpRequests.java]: Updated.

**Comment by chtyim** [comment]
LGTM

**Pull request merged**